### DISCUSSION TITLE
Interfacing FOX KH8 to HA via Waveshare 485 converter to Ethernet

### DISCUSSION THREAD

**Discussion by ianhills1000** [TonyM1958/HA-FoxESS-Modbus#13] [Q&A]
This is relatively new to myself and looking for some help please. 

I am trying to get my recently installed FoxESS KH8 inverter to communicate with HA. I have my Waveshare RS485 to Ethernet converter working on the network and HA installed on my Raspberry Pi 4. It would be good to discuss with someone that has actually done this with this KH8 model to know this will actually work. 

TonyM1958 modbusKH_RS485.yaml code looks the most promising and most recent though I do have some queries around this that I am not sure about on this. 

Is there any one that can give me a few appreciated tips please?

**Comment by TonyM1958** [comment]
Hi,

Yes, there are people using this integration with KH inverters. You can also use this: https://github.com/nathanmarlor/foxess_modbus

There is info on configuring your adapter here: https://github.com/nathanmarlor/foxess_modbus/wiki/Waveshare-RS485-to-ETH-%28B%29

Whichever integration you use, you need the IP address of the adapter to setup your HA configuration.

To configure this integration, follow the instructions here: https://github.com/TonyM1958/HA-FoxESS-Modbus/wiki/Installing-and-Configuring-HA-FoxESS-Modbus#configure-the-integration
